bundles/org.openhab.binding.flume/src/main/java/org/openhab/binding/flume/internal/handler/FlumeDeviceHandler.java

```diff
@@ -224,91 +242,123 @@ public void handleApiException(Exception e) {
     }
 
     /**
-     * Pools together several usage queries based on whether the value is expired and whether a channel is linked. Also,
-     * if necessary will update the usage from beginning to start of year so subsequent cumulative queries only need to
-     * ytd. Will update the values in the ExpiringCache as necessary.
-     *
-     * @throws FlumeApiException
-     * @throws IOException
-     * @throws InterruptedException
-     * @throws TimeoutException
-     * @throws ExecutionException
+     * {@link tryUpdateCumulativeStore} will query annual usage from HISTORICAL_YEAR_START to the end of last year.
+     * Flume
+     * limits queries to a duration of 1 year, so these must be broken up to individual queries. The result is stored
+     * so that subsequent startups don't have to requery.
      */
-    protected void tryQueryUsage(boolean forceUpdate)
+    protected void tryUpdateCumulativeStore()
             throws FlumeApiException, IOException, InterruptedException, TimeoutException, ExecutionException {
         @Nullable
         List<HashMap<String, List<FlumeApiQueryBucket>>> result;
-
-        boolean imperialUnits = isImperial();
-
+        List<FlumeApiQueryWaterUsage> listQuery = new ArrayList<FlumeApiQueryWaterUsage>();
         LocalDateTime now = LocalDateTime.now();
 
-        List<FlumeApiQueryWaterUsage> listQuery = new ArrayList<FlumeApiQueryWaterUsage>();
+        // Get sum of all historical full year readings only when binding starts or its the start of a new year
+        // This is to reduce the time it takes on the periodic queries. This will only happen when you run the binding
+        // the first time or .
+        boolean updateBaselineYears = cumulativeStore.lastUpdate() == LocalDateTime.MIN
+                || Duration.between(now, cumulativeStore.lastUpdate()).toDays() > 365;
```

Comment:
Did you consider a leap year?

Reply:
Yeah, that is fine. The purpose here is only to reset the baseline if the binding (for some reason like it was stopped) was not running for awhile. It will then reset run the queries.  The 365 is somewhat arbitrary and the rest of the logic is not dependent on that.